lib/vertica/protocol/frontend/startup.rb

```diff
@@ -8,12 +8,22 @@ def initialize(user, database, options = nil)
         @user     = user
         @database = database
         @options  = options
+        @type     = "vertica-rb"
+        @pid      = Process.pid.to_s
+        @platform = Gem::Platform.local.to_s
```

Comment:
My commit consists of StackOverflow answers I found for "how to get pid in ruby" and "how to get os in ruby".

I'll make it better.